Calculator/Calculator/Controller/CalculatorViewController.swift

```diff
@@ -14,149 +14,175 @@ class CalculatorViewController: UIViewController {
-    @IBOutlet weak var listStackView: UIStackView!
-    
-    var operandNumber = CalculatorValue.emptyArray
-    var formulaNumber = CalculatorValue.emptyArray
     
+    var currentOperand = CalculatorValue.emptyArray
+    var currentOperator = CalculatorValue.emptyArray
+    var inputString = CalculatorValue.emptyArray
+    var resultString = CalculatorValue.emptyArray
+        
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        initializeUI()
+        resetUI()
     }
-    
-    private func initializeUI() {
-        operandLabel.text = CalculatorValue.zero
-        operatorLabel.text = CalculatorValue.emptyArray
-        scrollStackView.arrangedSubviews.forEach { $0.removeFromSuperview() }
+  
+    private func resetUI() {
+        if currentOperand.isEmpty {
+            operandLabel.text = CalculatorValue.zero
+        } else {
+            operandLabel.text = currentOperand
+        }
+        
+        operatorLabel.text = currentOperator
     }
     
     @IBAction private func numberButtonDidTap(_ sender: UIButton) {
         guard let number = sender.title(for: .normal) else { return }
         
-        if operandNumber.count < 20 {
-            operandNumber += number
-            operandLabel.text = numberFormatter(operand: operandNumber)
-        } else if operandNumber.contains(Character(CalculatorValue.decimalPoint)) && operandNumber.count < 20 {
-            operandNumber += number
-            operandLabel.text = operandNumber
+        if currentOperand.count < 20 {
+            currentOperand += number
+        
+            operandLabel.text = currentOperand.numberFormatter()
         }
     }
     
     @IBAction private func dotButtonDidTap(_ sender: UIButton) {
-        if operandNumber.isEmpty {
-            operandNumber = CalculatorValue.zeroDecimal
-            operandLabel.text = numberFormatter(operand: operandNumber)
-        } else if operandNumber.last == Character(CalculatorValue.decimalPoint) || operandNumber.count == 20 || operandNumber.contains(Character(CalculatorValue.decimalPoint)) {
-            operandLabel.text = numberFormatter(operand: operandNumber)
+        guard currentOperand.count < 20 &&
+                !currentOperand.contains(Character(CalculatorValue.decimalPoint)) else { return }
+   
+        if currentOperand.isEmpty || currentOperand == CalculatorValue.minus {
+            currentOperand += CalculatorValue.zeroDecimal
         } else {
-            operandNumber += CalculatorValue.decimalPoint
-            operandLabel.text = operandNumber
+            currentOperand += CalculatorValue.decimalPoint
         }
+        
+        operandLabel.text = currentOperand
     }
     
-    @IBAction private func operatorButtonDidTap(_ sender: UIButton) {
-        guard let operatorSymbol = sender.title(for: .normal) else { return }
+    private func addListStackView() -> UIStackView {
+        let operatorLabel = UILabel()
```

Comment:
위 내용의 연장선으로 가독성을 위해 클로저를 활용하여 인스턴스를 만들어주기도 합니다.
참고만 해주세요 😊

```suggestion
        let operatorLabel: UILabel = {
            let label = UILabel()
            label.textColor = .white
            label.font = .preferredFont(forTextStyle: .title3)
            
            return label
        }()
```